test/hotspot/jtreg/serviceability/jvmti/events/FramePop/ClearAllFramePops/libClearAllFramePops.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include "jvmti.h"
+#include "jvmti_common.hpp"
+
+
+extern "C" {
+
+static jvmtiEnv *jvmti;
+static jvmtiEventCallbacks callbacks;
+static jrawMonitorID event_lock;
+static jboolean watch_events = JNI_FALSE;
+static int pop_count;
+static const char* TEST_THREAD_NAME_BASE = "Test Thread";
+static const char* TEST_CLASS_SIG = "LClearAllFramePops$TestTask;";
+
+static
+bool isTestThread(JNIEnv *jni, jvmtiEnv *jvmti, jthread thr) {
+  jvmtiThreadInfo inf = get_thread_info(jvmti, jni, thr);
+  bool result = strncmp(inf.name, TEST_THREAD_NAME_BASE, strlen(TEST_THREAD_NAME_BASE)) == 0;
+  deallocate(jvmti, jni, inf.name);
+
+  return result;
+}
+
+static
+void printInfo(JNIEnv *jni, jvmtiEnv *jvmti, jthread thr, jmethodID method, int depth) {
+  jclass cls;
+  char *mname, *msig, *csig;
+  jvmtiThreadInfo inf = get_thread_info(jvmti, jni, thr);
+
+  check_jvmti_status(jni, jvmti->GetMethodDeclaringClass(method, &cls), "Error in GetMethodDeclaringClass.");
+  check_jvmti_status(jni, jvmti->GetClassSignature(cls, &csig, nullptr), "Error in GetClassSignature.");
+  check_jvmti_status(jni, jvmti->GetMethodName(method, &mname, &msig, nullptr), "Error in GetMethodName.");
+
+  LOG(" %s: %s.%s%s, depth = %d\n", inf.name, csig, mname, msig, depth);
+
+  deallocate(jvmti, jni, inf.name);
+  deallocate(jvmti, jni, mname);
+  deallocate(jvmti, jni, msig);
+  deallocate(jvmti, jni, csig);
+}
+
+void JNICALL MethodEntry(jvmtiEnv *jvmti, JNIEnv *jni,
+                         jthread thr, jmethodID method) {
+  RawMonitorLocker rml(jvmti, jni, event_lock);
+
+  if (watch_events == JNI_FALSE) {
+    return;
+  }
+  if (!isTestThread(jni, jvmti, thr)) {
+    return; // not a tested thread
+  }
+  jclass cls;
+  char *csig;
+  jvmtiThreadInfo inf = get_thread_info(jvmti, jni, thr);
```

Comment:
Looks like this is not needed (and inf.name not deallocated)

Reply:
Good catch. Removed the line now.